mobilecoind-json/src/bin/main.rs

```diff
@@ -267,6 +288,79 @@ fn transfer(
     Ok(Json(JsonTransferResponse::from(&resp)))
 }
 
+/// Performs a transfer from a monitor and subaddress. The public keys and amount are in the POST data.
+#[post(
+    "/monitors/<monitor_hex>/subaddresses/<subaddress_index>/generate-tx",
+    format = "json",
+    data = "<generate>"
+)]
+fn generate_tx(
+    state: rocket::State<State>,
+    monitor_hex: String,
+    subaddress_index: u64,
+    generate: Json<JsonGenerateTxRequest>,
+) -> Result<Json<JsonGenerateTxResponse>, String> {
+    let monitor_id =
+        hex::decode(monitor_hex).map_err(|err| format!("Failed to decode monitor hex: {}", err))?;
+
+    let public_address = PublicAddress::try_from(&generate.transfer.receiver)?;
+
+    // Generate an outlay
+    let mut outlay = mc_mobilecoind_api::Outlay::new();
+    outlay.set_receiver(public_address);
+    outlay.set_value(
+        generate
+            .transfer
+            .value
+            .parse::<u64>()
+            .map_err(|err| format!("Failed to parse amount: {}", err))?,
+    );
+
+    let mut inputs: Vec<mc_mobilecoind_api::UnspentTxOut> = Vec::new();
+    for input in generate.input_list.iter() {
```

Comment:
This can be done with collect(): (something like this)
```
let inputs:  Vec<mc_mobilecoind_api::UnspentTxOut> = generate.input_list.iter().map(|input| mc_mobilecoind_api::UnspentTxOut::try_from(input).map_err(|err| format!("Failed to convert input {:?}: {}", input, err)))).collect::<Result<_, String>>()?;
```